content/articles/beginner's-guide-to-lottie-animations-in-android-using-kotlin/index.md

```diff
@@ -0,0 +1,388 @@
+
+animationView.setAnimation("hello-world.json")
+
+animationView.loop(true)
+
+animationView.playAnimation()
+```
+- This will load the file and parse the animation .
+
+  
+
+# Functionality
+
+Because I have used an adapter to display the contents of the animation, the functionality is in the adapter, in the `bind` function
+
+```kotlin
+
+fun bind(onBoardingPage: OnboardingPage) {
+
+val res = root.context.resources
+
+root.titleTv?.text = res.getString(onBoardingPage.titleResource)
+
+root.subTitleTv?.text = res.getString(onBoardingPage.subTitleResource)
+
+root.descTV?.text = res.getString(onBoardingPage.descriptionResource)
+
+root.img.setAnimation(onBoardingPage.logoResource)
+
+root.img. playAnimation()
+
+  
+
+}
+
+```
+
+`.setAnimation`, sets the raw animations to the image view
+
+`.playAnimation()`, plays the animation
+
+  
+
+# 4. MainActivity
+
+If you use the mainActivity to add a function, you will have to create a function to set up the animation.
+
+  
+
+```kotlin
+
+fun setupAnimation(){
+
+val animation = findViewById<LottieAnimationView>(R.id.progressBar)
+
+animation.speed = 2.0F // How fast does the animation play
+
+animation.progress = 50F // Starts the animation from 50% of the beginning
+
+animation.addAnimatorUpdateListener {
+
+// Called every time the frame of the animation changes
+
+}
+
+animation.repeatMode = LottieDrawable.RESTART // Restarts the animation (you can choose to reverse it as well)
+
+animation.cancelAnimation() // Cancels the animation
+
+}
+
+```
+
+You shall use;
+
+`.speed`- for setting how fast the animation plays
+
+-`.progress`- to choose where the animation starts
+
+-`repeatMode` - restarts the animation
+
+  
+  
+
+# The error I ran into
+
+When using a class that holds the animation files like this,
+
+```kotlin
+
+enum class OnboardingPage(
+
+@StringRes val titleResource: Int,
+
+@StringRes val subTitleResource: Int,
+
+@StringRes val descriptionResource: Int,
+
+@DrawableRes val logoResource: Int
+
+) {
+
+  
+
+ONE(R.string.onboarding_slide1_title, R.string.onboarding_slide1_subtitle,R.string.onboarding_slide1_desc, R.raw.comp),
+
+TWO(R.string.onboarding_slide2_title, R.string.onboarding_slide1_subtitle,R.string.onboarding_slide1_desc, R.raw.jump),
+
+THREE(R.string.onboarding_slide3_title, R.string.onboarding_slide1_subtitle,R.string.onboarding_slide1_desc, R.raw.cycling)
+
+  
+
+}
+
+```
+
+you might come across a lint error in the adapter,on this line,`root.img.setAnimation(onBoardingPage.logoResource)`. If you add the `@SuppressLint` annotation, the project will not run. The error does not necessarily affect the functionality of the app.
+
+  
+
+# Output
+
+<iframe width="478" height="269" src="https://www.youtube.com/embed/Sw6gz27DLFs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+  
```

Comment:
Maybe you can also have the tutorial's APK . You can host it on Google drive and get a sharable link.